plugins/modules/gconftool2.py

```diff
@@ -160,75 +102,199 @@ def main():
-    dummy, current_value = gconf_pref.call("get", fail_onerr=False)
-
-    # Check if the current value equals the value we want to set.  If not, make
-    # a change
-    if current_value != value:
-        # If check mode, we know a change would have occurred.
-        if module.check_mode:
-            # So we will set the change to True
-            change = True
-            # And set the new_value to the value that would have been set
-            new_value = value
-        # If not check mode make the change.
-        else:
-            change, new_value = gconf_pref.call(state)
-    # If the value we want to set is the same as the current_value, we will
-    # set the new_value to the current_value for reporting
-    else:
-        new_value = current_value
-
-    facts = dict(gconftool2={'changed': change,
-                             'key': key,
-                             'value_type': value_type,
-                             'new_value': new_value,
-                             'previous_value': current_value,
-                             'playbook_value': module.params['value']})
-
-    module.exit_json(changed=change, ansible_facts=facts)
+    def __init_module__(self):
+        self.runner = gconftool2_runner(self.module, check_rc=True)
+        if self.vars.state != "get":
+            if not self.vars.direct and self.vars.config_source is not None:
+                self.module.fail_json(msg='If the "config_source" is specified then "direct" must be "true"')
+
+        self.vars.set('previous_value', self._get(), fact=True)
+        self.vars.set('value_type', self.vars.value_type)
+        self.vars.set_meta('value', initial_value=self.vars.previous_value)
+        self.vars.set('playbook_value', self.vars.value, fact=True)
+
+    def _make_process(self, fail_on_err):
+        def process(rc, out, err):
+            if err and fail_on_err:
+                self.ansible.fail_json(msg='gconftool-2 failed with error: %s' % (str(err)))
+            self.vars.value = out.rstrip()
+            return self.vars.value
+        return process
+
+    def _get(self):
+        return self.runner("state key", output_process=self._make_process(False)).run(state="get")
+
+    def state_get(self):
+        self.deprecate(
+            msg="State 'get' is deprecated. Please use the module community.general.gconftool2_info instead",
+            version="8.0.0", collection_name="community.general"
+        )
+
+    def state_absent(self):
+        with self.runner("state key", output_process=self._make_process(False)) as ctx:
+            ctx.run()
+        self.vars.set('new_value', None, fact=True)
+
+    def state_present(self):
+        with self.runner("direct config_source value_type state key value", output_process=self._make_process(True)) as ctx:
+            self.vars.set('new_value', ctx.run(), fact=True)
+
+
+# class GConf2Preference(object):
+#     def __init__(self, ansible, key, value_type, value,
```

Comment:
Does this huge block of commented out code need to stay?

Reply:
oopsie, forgot to remove that